### PR TITLE
Fixes for VM CPU topology

### LINKED ISSUE #9456: vm: SMT not working
I need to test scheduler patches for SMT but none of my VMs seem to have SMT supported. Here's my host `lscpu` layout:
```
Architecture:                    x86_64
CPU op-mode(s):                  32-bit, 64-bit
Byte Order:                      Little Endian
Address sizes:                   39 bits physical, 48 bits virtual
CPU(s):                          8
On-line CPU(s) list:             0-7
Thread(s) per core:              2
Core(s) per socket:              4
Socket(s):                       1
NUMA node(s):                    1
Vendor ID:                       GenuineIntel
CPU family:                      6
Model:                           142
Model name:                      Intel(R) Core(TM) i5-8350U CPU @ 1.70GHz
Stepping:                        10
CPU MHz:                         1900.000
CPU max MHz:                     3600,0000
CPU min MHz:                     400,0000
BogoMIPS:                        3799.90
Virtualization:                  VT-x
L1d cache:                       128 KiB
L1i cache:                       128 KiB
L2 cache:                        1 MiB
L3 cache:                        6 MiB
NUMA node0 CPU(s):               0-7
Vulnerability Itlb multihit:     KVM: Mitigation: VMX disabled
Vulnerability L1tf:              Mitigation; PTE Inversion; VMX conditional cache flushes, SMT vulnerable
Vulnerability Mds:               Mitigation; Clear CPU buffers; SMT vulnerable
Vulnerability Meltdown:          Mitigation; PTI
Vulnerability Spec store bypass: Mitigation; Speculative Store Bypass disabled via prctl and seccomp
Vulnerability Spectre v1:        Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:        Mitigation; Full generic retpoline, IBPB conditional, IBRS_FW, STIBP conditional, RSB filling
Vulnerability Srbds:             Mitigation; Microcode
Vulnerability Tsx async abort:   Mitigation; TSX disabled
Flags:                           fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb invpcid_single pti ssbd ibrs ibpb stibp tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust sgx bmi1 avx2 smep bmi2 erms invpcid mpx rdseed adx smap clflushopt intel_pt xsaveopt xsavec xgetbv1 xsaves dtherm ida arat pln pts hwp hwp_notify hwp_act_window hwp_epp md_clear flush_l1d
```

### PR DESCRIPTION
**Pull request by stgraber**
Closes #9456

### CODE REVIEW

File: doc/instances.md [hunk 1]
```diff
@@ -186,6 +186,37 @@ instance, relative to any other instance which is using the same CPU(s).
 scheduler priority score when a number of instances sharing a set of
 CPUs have the same percentage of CPU assigned to them.
 
+### VM CPU topology
+LXD virtual machines default to having just one vCPU allocated which
+shows up as matching the host CPU vendor and type but has a single core
+and no threads.
+
+When `limits.cpu` is set to a single integer, this will cause multiple
+vCPUs be allocated and exposed to the guest as full cores. Those vCPUs
```
Comment (tomponline) [inline doc/instances.md:195]: "to be"?

Comment (stgraber) [reply doc/instances.md]: Fixed!

**Review by tomponline — APPROVED** [review]

**Review by brauner — APPROVED** [review]
I feel we should have had separate keys:
1. `limits.cpu`
2. `limits.cpu.pin`
or sm.

**Comment by stgraber** [comment]
@brauner a bit hard to change that after years of `limits.cpu` working in this way. The VM behavior actually lines up with the container behavior with the exception of the faking NUMA and core/thread topology which isn't really a thing for containers.

I think we have a better chance to make things easier by having the doc updated (initial pass in this PR) and maybe exposing more details on topology in things like `lxc info` as well as adding a way to show warnings on startup to the user.

**Comment by brauner** [comment]
> @brauner a bit hard to change that after years of `limits.cpu` working in this way. The VM behavior actually lines up with

I wasn't suggesting we change behavior of the existing key. Just felt that we could've thought about this back when we introduced `limits.cpu`. I feel that `limits.cpu` right now expresses two distinct concepts in a way.
